geoserve/geoserve.go

```diff
@@ -4,25 +4,33 @@ import (
 	"compress/gzip"
 	"encoding/json"
 	"fmt"
-	geoip2 "github.com/oschwald/geoip2-golang"
 	"io/ioutil"
-	"log"
 	"net"
 	"net/http"
 	"os"
 	"strings"
 	"time"
+
+	"github.com/getlantern/golog"
+	"github.com/golang/groupcache/lru"
+	geoip2 "github.com/oschwald/geoip2-golang"
 )
 
 const (
 	DB_URL = "https://geolite.maxmind.com/download/geoip/database/GeoLite2-City.mmdb.gz"
+
+	CacheSize = 50000
```

Comment:
We can tweak this to achieve the right balance between memory consumption and CPU usage.  Basically, the cache helps avoid a little computation to look up the geolocation information, and a lot of computation to generate the JSON representation of the result.5